src/firefly/js/visualize/Histogram.jsx

```diff
@@ -27,16 +29,55 @@ var Histogram = React.createClass(
 
         getDefaultProps() {
             return {
-                data: undefined,
-                source: undefined,
-                height: 300,
-                logs: undefined,
-                reversed: undefined,
                 desc: 'Sample Distribution',
                 binColor: '#d1d1d1'
             };
         },
 
+        shouldComponentUpdate(nextProps) {
```

Comment:
this is purely my opinion without studying the code.
there is a lot of logic in this check.  it would be helpful if there is a short comments outlining when this component should update.